tensorboard/components/tf_tensorboard/tf-tensorboard.html

```diff
@@ -201,131 +181,221 @@
     import {setRouter, createRouter} from "../tf-backend/router.js";
     import {fetchRuns} from "../tf-backend/runsStore.js";
 
+    const COMPONENTS = {
+      'scalars': 'tf-scalar-dashboard',
+      'images': 'tf-image-dashboard',
+      'audio': 'tf-audio-dashboard',
+      'graphs': 'tf-graph-dashboard',
+      'distributions': 'tf-distribution-dashboard',
+      'histograms': 'tf-histogram-dashboard',
+      'embeddings': 'vz-projector-dashboard',
+      'text': 'tf-text-dashboard',
+    };
+    if (!_.isEqual(Object.keys(COMPONENTS), TABS)) {
+      throw new Error(
+        `Bad set of components: ${Object.keys(COMPONENTS)} vs. ${TABS}`);
+    }
+
     Polymer({
       is: "tf-tensorboard",
       behaviors: [AutoReloadBehavior],
       properties: {
+        /**
+         * We accept a router property only for backward compatibility:
+         * setting it triggers an observer that simply calls
+         * `setRouter`.
+         */
         router: {
-          type: Object,  // only to trigger an observer
+          type: Object,
           observer: '_updateRouter',
         },
+
+        /**
+         * If this is set to a string, TensorBoard will switch to "demo mode"
```

Comment:
I'll deprecate it: it's still used in tests (`fastTabSwitch.ts`, `e2eTests.ts`).